hypothesis-python/src/hypothesis/searchstrategy/collections.py

```diff
@@ -146,6 +153,18 @@ def do_draw(self, data):
         while elements.more():
             value = data.draw(self.element_strategy)
             k = self.key(value)
+
+            draw_length = data.blocks.last_block_length * 8
+            while indices and k in seen:
```

Comment:
This could use a quick comment indicating that it only runs when we have special knowledge of the values produced by `element_strategy`.

(It doesn't have to be detailed; just enough that the reader has a basic understanding instead of having to track down `indices`.)